File: lane.py
import cv2
import numpy as np


img = cv2.imread("./testimages/im2345.jpg")
img_grey = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
crop_img = img_grey[399:, :]
print(np.shape(crop_img))
cv2.imshow("cropped", crop_img)
border_size = 4

dims = np.shape(crop_img)
img = np.ones([(dims[0]+2*border_size)*10, (dims[1]+2*border_size)*10], dtype=np.uint8)

img[:dims[0], :dims[1]] = crop_img


imedge = cv2.Canny(img, 250, 700, 3)

size = border_size * 8

imedge[:size, :size] = 0
imedge[dims[0]-size:dims[1], :size] = 0
imedge[1:size, dims[0]-size:dims[1]] = 0
imedge[dims[0]-size:dims[0], dims[1]-size:dims[1]] = 0

cv2.imshow("Epic shit!", imedge)


cv2.waitKey(100000)

File: lane.m
clear all
im = imread('testimages/im5.jpg');


imgrey = rgb2gray(im);


imcrop = imgrey(400:end, 1:end);
imshow(imcrop);

bordersize = 4
img = uint8(ones(size(imcrop)+[bordersize*2, bordersize*2])*10);
img(bordersize+1:end-bordersize, bordersize+1:end-bordersize) = imcrop; 


imshow(img);



imedge = edge(img,'Canny',[0.08, 0.3], 3);


%corner hack
size = bordersize * 8;
imedge(1:size, 1:size) = 0;
imedge(end-size:end, 1:size) = 0;
imedge(1:size, end-size:end) = 0;
imedge(end-size:end, end-size:end) = 0;




se = strel('disk',4);
imedge = imdilate(imedge,se);
figure, imshow(imedge)


imedge = imfill(imedge, 'holes');
se = strel('disk',10);
imedge = imopen(imedge,se);


figure,imshow(imedge)

imsmooth = imgaussfilt(uint8(imedge)*255,20);
figure,imshow(imsmooth)

%s = regionprops((img_filled), 'centroid','eccentricity','Area', 'FilledImage', 'FilledArea','Perimeter', 'Solidity', 'PixelIdxList');


